tracing/src/main/java/com/palantir/remoting1/tracing/AsyncSlf4jSpanObserver.java

```diff
@@ -56,11 +56,11 @@ static ZipkinCompatibleSerializableSpan fromSpan(Span span) {
                     .name(span.getOperation())
                     .parentId(span.getParentSpanId())
                     .timestamp(span.getStartTimeMs())
-                    .duration(nanoToMilli(span.getDurationNs()))  // Zipkin-durations are micro-seconds, rounded up
+                    .duration(nanoToMicro(span.getDurationNs()))  // Zipkin-durations are micro-seconds, rounded up
                     .build();
         }
 
-        static long nanoToMilli(long nano) {
+        static long nanoToMicro(long nano) {
             return (long) Math.ceil(nano / (double) 1000);
```

Comment:
Why are we doing double math here? This should be:

`return Math.max(1L, nano / 1000L);`